docs(pie-docs): update engineer and foundation pages copy (#343)
justeattakeaway/pie: apps/pie-docs/src/foundations/iconography/overview.md

## Icon anatomy

> Our icons are always placed inside a bounding box, which ensures that vertical and horizontal spacing always adhere to the guidance shown in our [spacing guides](/foundations/spacing/overview/).

Reviewer: ```suggestion
Our icons are always placed inside a bounding box, which ensures that vertical and horizontal spacing always adheres to our [spacing guides](/foundations/spacing/overview/).
```

Reviewer 2: 👍

---
## Using icons in your products

> Choosing and using icons for your products isn’t an easy task. Especially if we consider that an excellent icon can extract the most important visual characteristics of a product or idea in order to make the depiction instantly recognisable.

Reviewer: ```suggestion
Choosing and using icons for your products isn’t an easy task. An excellent icon can extract the most important visual characteristics of a product or idea, making the depiction instantly recognisable.
```

Reviewer 2: 👍
